[blog] 2020 in review and beyond (#24130)
mui-org/material-ui: docs/pages/blog/2020.md

date: 2021-01-01T00:00:00.000Z
authors: ['oliviertassinari', 'mbrookes']
card: true
---

2020 has been another great year, not only for Material-UI, but also for the ecosystem.
We have taken on ambitious challenges and started to scale the project.

## Growth

- 📦 From 3.2M to [6.4M](https://npm-stat.com/charts.html?package=%40material-ui%2Fcore&from=2019-11-30&to=2020-12-31) downloads per month (from 14% to 18% share of the React ecosystem).
- 📈 From 3.1M to 4.6M unique visitors per year on the documentation.
- ⭐️ From 53.3k to 63.8k stars, leave us [yours 🌟](https://github.com/mui-org/material-ui).
- 👨‍👩‍👧‍👦 From 1,581 to [2,052](https://github.com/mui-org/material-ui/graphs/contributors) contributors.
- 💰 Grew financial support by 2.1X in 2020, compared to 2019.
- 🏢 From 3 to 6 full-time equivalent developers, spread among multiple financially supported [core team members](https://material-ui.com/discover-more/team/).

The numbers speak for themselves. We have reproduced the [traction we got in 2019](/blog/2019/).

## In review

When we started 2020, we were celebrating the launch of many new components.

> We were busy building features to match feature parity with Ant Design, and more importantly, acting on the top feedback we got doing a [first user survey](/blog/2019-developer-survey-results/).

Reviewer: ```suggestion
We were busy building features to match feature parity with Ant Design, and more importantly, acting on the top feedback we got doing the [first user survey](/blog/2019-developer-survey-results/).
```

We have started to leverage these trends as opportunities in the next version of the library: v5.

## Achievements

We have achieved most of what we could have hoped for.

- We have welcome 3 new members in the company: [Damien](/blog/spotlight-damien-tassone/), [Marija](/blog/marija-najdova-joining/), and [Danail](/blog/danail-hadjiatanasov-joining/).
- We have made significant progress with [v5](https://next.material-ui.com/). We have made half the breaking changes planned. We have migrated our [first component](https://next.material-ui.com/components/slider/) to the new style architecture (emotion by default, adapter for styled-components, and unstyled).
- We have introduced new components (some in the core, some in the lab):
  - [Alert](/components/alert/)
  - [DataGrid](/components/data-grid/)
  - [DatePicker](https://next.material-ui.com/components/date-picker/)
  - [LoadingButton](https://next.material-ui.com/components/buttons/#loading-buttons)
  - [Timeline](/components/timeline/)
  - [TrapFocus](https://next.material-ui.com/components/trap-focus/)

> - We have fixed most of the issues with the [autocomplete](/components/autocomplete/). We have received an overwhelming interest in the component. It was impressive to see.

Reviewer: ```suggestion
- We have fixed most of the issues with the [Autocomplete](/components/autocomplete/). We have received an overwhelming interest in the component. It was impressive to see.
```

I would expect component names to be always capitalized

- We have completed the work for [strict mode](https://reactjs.org/docs/strict-mode.html) support.
- We have increased the adoption of TypeScript in the codebase. We don't plan a dedicated migration but to write new code in TypeScript, as we go.
- We have migrated most of the tests from Enzyme to [Testing Library](https://testing-library.com/).
- We have modernized the system, introducing an [`sx` prop](https://next.material-ui.com/system/basics/) to be used in all the core components.
- We have added support for [Figma](/store/items/figma-react/) and [Adobe XD](/store/items/adobe-xd-react/).
- We have released the first Enterprise component in an alpha version: [XGrid](/components/data-grid/#commercial-version).

## Looking at 2021

2020 was great, 2021 is going to be even more exciting!
We will continue in the same direction, it's still [**day one**](https://www.sec.gov/Archives/edgar/data/1018724/000119312517120198/d373368dex991.htm). Here is a breakdown of our [roadmap](/discover-more/roadmap/).

### Branding

Up until now, we never had a mindful reflection on what the branding of Material-UI should be. We picked a color from the 2014 Material Design palette, and that's it.

> The material-ui.com marketing website will soon wear a brand new style! We need to create our own identity, distinct from Material Design.

Reviewer: ```suggestion
The [material-ui.com](https://material-ui.com/) marketing website will soon wear a brand new style! We need to create our own identity, distinct from Material Design.
```

- Reduce bundle size: split IE11 into a different bundle [-6kB](https://github.com/mui-org/material-ui/pull/22814#issuecomment-700995216), migrate to emotion [-5kB](https://github.com/mui-org/material-ui/pull/23308#issuecomment-718748835), Popper.js v2 upgrade [-700B](https://github.com/mui-org/material-ui/pull/21761#issuecomment-657135498).
- Improve the documentation website: search shortcut, page rating, fast material icons copy button, etc.

And [much more](https://github.com/mui-org/material-ui/issues/20012).

### A second theme

While Material Design is a predominant design choice for building rich client-side applications, it's not the only popular design trend. We plan to provide a [second theme](https://github.com/mui-org/material-ui/issues/22485) to expand the range of use cases for Material-UI, striking a balance between:

- alignment with the best-looking design trends of hot startups in the US.
- optimization of its usage for rich client-side applications while keeping it good enough for landing pages.
- simplicity of the design for allowing simpler customizations.
- complementarity with Material Design.

> It will be built on the unstyled components.

Reviewer: ```suggestion
It will be built on top of the unstyled components.
```

### Unstyled

While the completion of the unstyled components was originally part of the v5 milestone,
we will likely finish this effort independently.
Outside of the requirement to introduce breaking changes on the component customization API, e.g. from `PaperProps` to `componentsProps`, [RFC #20012](https://github.com/mui-org/material-ui/issues/21453), we can work on unstyled at the same time we make progress with the second theme. The two efforts should go hand in hand.

### Scale

If we sustain the current growth rate, we should be able to double the size of the team, from 6 to over 10 members in the company without raising. While we will focus on strengthening all the efforts that we have started, it should start to give us room to take on significantly more ambitious problems guided by the following foundations:

- React dominance in the UI development landscape will increase and stay for a very long time.
- Developers are looking to build faster and more appealing UIs.

> - Low-code will progressively become mainstream for professional developers and knowledge workers.

Reviewer: It looks like the bog finishes suddenly without any conclusion. Would propose finishing up with `Hiring` section, similar to how it was done on the 2019 blog post (if applicable :D)

Author: Yeah, agree, the end is too abrute!

---
title: 2020 in review and beyond

> description: 2019 was a great year for Material-UI. It puts us on an exciting path to solve even greater challenges in the coming years!

Author: ```suggestion
description: 2020 has been another great year, not only for Material-UI, but also for the ecosystem.
```

date: 2021-01-01T00:00:00.000Z
authors: ['oliviertassinari', 'mbrookes']
card: true
---

2020 has been another great year not only for Material-UI but also for the ecosystem.
We have taken on ambitious challenges and started to scale the project.

## Growth

> - 📦 From 3.2M to [6.4M](https://npm-stat.com/charts.html?package=%40material-ui%2Fcore&from=2019-11-30&to=2020-12-31) downloads per month (from 14% to 18% of market share in the React ecosystem).

Author: I have computed it with https://docs.google.com/spreadsheets/d/1l5j3Xjtvm9XZtmb4ulLiWElQaXSlZlyCWT5ONrQMpBo/edit#gid=0, I got 18.6%  which I have floored.

> - 📦 From 3.2M to [6.4M](https://npm-stat.com/charts.html?package=%40material-ui%2Fcore&from=2019-11-30&to=2020-12-31) downloads per month (from 14% to 18% of market share in the React ecosystem).

Reviewer 2: Thanks for the link, but I still have no idea what you're measuring since you didn't say, and it's not clear from the npm-stat page what "material-ui" refers to, nor which date nor timeframe you're using.

But taking current weekly downloads of @material-ui/core vs react-dom, it is indeed over 18%.

- The more new components we add, the more developers want new ones. For instance, we rarely heard about the need for providing charts in 2019, it's now one of the top requested. In our last survey, we had 10% more requests for new components, even though, we brought new ones. We were surprised, it was counter-intuitive to us.
- The pain around the customization of the components is growing. We also frequently hear that there is a stiff learning curve for using `@material-ui/styles`. We are reaching a larger audience used to raw CSS without extensive knowledge of specificity.
- The demand for Material Design is stable, no longer growing. At the same time, the request for new ready-to-use themes is up. Our strategy around providing building blocks to build custom designed UI starts to gain traction.

We have started to leverage these trends as opportunities in the next version of the library: v5.

## Achievements

We have achieved most of what we could have hoped for.

- We have welcome 3 new members in the company: [Damien](/blog/spotlight-damien-tassone/), [Marija](/blog/marija-najdova-joining/), and [Danail](/blog/danail-hadjiatanasov-joining/).

> - We have made significant progress with [v5](https://next.material-ui.com/). We have done half the breaking changes planned. We have migrated our [first component](https://next.material-ui.com/components/slider/) to the new style architecture (emotion + styled-components + unstyled).

Author: ```suggestion
- We have made significant progress with [v5](https://next.material-ui.com/). We have done half the breaking changes planned. We have migrated our [first component](https://next.material-ui.com/components/slider/) to the new style architecture (emotion by default, adapter for styled-components, and unstyled).
```

- Improve customizability. Add new powers to the theme with dynamic color & [variant](https://next.material-ui.com/customization/typography/#adding-amp-disabling-variants) support. Add a new [`sx` prop](https://next.material-ui.com/system/basics/) for quick customizations to all the components. Expose global class names. Deprecate the `makeStyles` and `withStyles` API.
- Breaking changes on the API to make it more intuitive.
- Add full support for React strict mode. We recommend to enable it.
- Improve the performance of the system by a x3-x5 [factor](https://github.com/mui-org/material-ui/issues/21657#issuecomment-707140999).
- Reduce bundle size: split IE11 into a different bundle [-6kB](https://github.com/mui-org/material-ui/pull/22814#issuecomment-700995216), migrate to emotion [-5kB](https://github.com/mui-org/material-ui/pull/23308#issuecomment-718748835), Popper.js v2 upgrade [-700B](https://github.com/mui-org/material-ui/pull/21761#issuecomment-657135498).
- Improve the documentation website: search shortcut, page rating, fast material icons copy button, etc.

And [much more](https://github.com/mui-org/material-ui/issues/20012).

### A second theme

> While Material Design is a predominant design choice for building rich client-side applications, it's not in the landscape of landing and marketing pages.

Author: Why not a driver at all? Isn't what Bootstrap was mostly used for initially? I do agree that it might not be the first driver. How about I inverse the order between the two drivers?

> While Material Design is a predominant design choice for building rich client-side applications, it's not in the landscape of landing and marketing pages.

Reviewer 2: > Why not a driver at all?

For the reason given.

> While Material Design is a predominant design choice for building rich client-side applications, it's not in the landscape of landing and marketing pages.

Author: > most Material-UI components have no place on a home page, landing page, or marketing pages.

Are you saying the following components have no place in a landing page:

- Container
- Grid
- Box
- Tooltip
- Accordion
- Breadcrumbs
- Pagination
- Button
- Icons
- Typography
- Radio
- Select
- Input
- Rating

As well as the ones we have an interest in building in the coming months:

- NProgress
- Scrollspy
- Stack
- Carousel

?

> While Material Design is a predominant design choice for building rich client-side applications, it's not in the landscape of landing and marketing pages.

Reviewer 2: https://yourlogicalfallacyis.com/strawman

Given the relative simplicity of the majority of elements that make up a home page (most of the ones listed, if you exclude the clutching at straws) will continue to dominate that space.

Optimising the design of a large majority of the components, including the most complex, for use on a home page or landing page makes no sense.

> While Material Design is a predominant design choice for building rich client-side applications, it's not in the landscape of landing and marketing pages.

Reviewer 2: https://yourlogicalfallacyis.com/strawman

Given the relative simplicity of the majority of elements that make up a home page (most of the ones listed, if you exclude the clutching at straws) HTML & CSS will continue to dominate that space.

Optimising the design of all of the components, including the most complex, for use on a home page or landing page makes no sense, when only a small minority of the most simple have any reasonable application there, especially given the tiny minority of sites that might use React for their home page in the first place.

If  we want to follow the design trends of the latest startups, we need to look at their applications and dashboards, not their home pages, since it's where the vast majority of our components play.

> While Material Design is a predominant design choice for building rich client-side applications, it's not in the landscape of landing and marketing pages.

Author: > If we want to follow the design trends of the latest startups, we need to look at their applications and dashboards, not their home pages, since it's where the vast majority of our components play.

Ok, good for me, I expect they share the same design guidelines and components between the two landscape.

> This use case for React has yet to win a large-scale adoption by the community but the traction of Next.js is a strong signal that it has the potential to happen. React pushing [server-side components](https://reactjs.org/blog/2020/12/21/data-fetching-with-react-server-components.html), with zero bundle size overhead could accelerate it.

Reviewer 2: What's the use-case for server side components on a home page, landing page, or marketing pages, that will cause an upswing in the adoption of React for this use-case?

Isn't it fair to say that HTML and CSS are a far better fit, and always will be?

Author: > Isn't it fair to say that HTML and CSS are a far better fit, and always will be?

I think that there is little place for JavaScript in there. Its usage should be limited, with a couple of carousel, tooltip, scrollspy, menu, popup, slider, or combo box. The new server-side components deliver zero JavaScript bundle size, it's effectively HTML and CSS. It can replace slim (rails), twig (Symfony), blade (laravel), etc. However, unlike jQuery style, you get components with isolated client-side logic for the few places where you need it.

IMHO, Next.js wouldn't be growing if HTML **and** CSS raw was great (Gatsby is losing over Next.js)

Author: https://twitter.com/rauchg/status/1342963266004258817/photo/1 😆

Reviewer 2: Okay, sound like we agree. Optimising the second theme for home page usage doesn't make much sense.

Author: > Okay, sound like we agree. Optimising the second theme for home page usage doesn't make much sense.

In our survey, 6 months ago, we go about 20% of the usage of React [for landing pages](https://material-ui.com/blog/2020-developer-survey-results/#16-what-delivery-mechanism-are-you-using). I agree that it shouldn't be the primary factor to optimize with.

However, considering the 20%, that it's growing, that server-side components, might make it explodes in the next 5 years and that Material Design is weak in this segment, I think that we should give it a low weight but still make sure that it works.

Reviewer 2: I don't know how you anticipate optimising the design of the majority of components for use on the homepage when only a minority, many of which have no visible design aspect, have any use there.

> At the same time, the design trend of the latest hot startups isn't close to Material Design.

Author: I doubt there is a downward trend with Material Design. It seems to be stable.

https://www.shutterstock.com/ is the biggest flagship I know of using Material Design for a high traffic landing page like website. I put e-commerce, blog, portfolio in the same category.

> At the same time, the design trend of the latest hot startups isn't close to Material Design.

Reviewer 2: I doubt there is a downward trend with Material Design.

The point is that since Material Design never had much of a place on the home page ("not in the landscape" in your words), it shouldn't come as a surprise that "the design trend of the latest hot startups isn't close to Material Design", since it never was, still isn't, and never will be.

We plan to provide a [second theme](https://github.com/mui-org/material-ui/issues/22485) to perfectly answer these two problems, striking a balance between:

- simplicity of the design for simpler customizations
- alignment with the best-looking design trends of hot startups

It will be built on the unstyled components.

### Unstyled

> While the completion of the unstyled components was originally part of the v5 milestone.

Reviewer 2: ```suggestion
While the completion of the unstyled components was originally part of the v5 milestone,
```

> We will likely finish this effort independently.

Reviewer 2: ```suggestion
we will likely finish this effort independently.
```

> Outside of the requirement to introduce breaking changes on the component customization API, e.g. from `PaperProps` to the `componentsProps`, [RFC #20012](https://github.com/mui-org/material-ui/issues/21453), we can work on unstyled at the same time we make progress with the second theme. The two efforts should go hand in hand.

Reviewer 2: ```suggestion
Outside of the requirement to introduce breaking changes on the component customization API, e.g. from `PaperProps` to `componentsProps`, [RFC #20012](https://github.com/mui-org/material-ui/issues/21453), we can work on unstyled at the same time we make progress with the second theme. The two efforts should go hand in hand.
```

### Scale

> If we sustain the current growth rate, we should be able to double the size of the team, from 6 to over 10 members in the company without raising. While we will focus on strengthening all the efforts that we have started, it should start to give us room to take on significantly more ambitious problems guided by the following foundations:

Reviewer 2: > from 6 to over 10

That isn't double 😄

Author: It's to leave more room for imagination 😇

- React dominance in the UI development landscape will increase and stay for a very long time.

> - Developers are looking for building faster and more appealing UIs.

Reviewer 2: ```suggestion
- Developers are looking to build faster and more appealing UIs.
```
